var/spack/repos/builtin/packages/abinit/package.py

```diff
@@ -90,52 +91,44 @@ class Abinit(Package):
     depends_on("hdf5+mpi", when='+mpi+hdf5')  # required for NetCDF-4 support
 
     # pin libxc version
-    depends_on("libxc@2.2.1")
-
-    def validate(self, spec):
-        """
-        Checks if incompatible variants have been activated at the same time
+    depends_on("libxc@2.2.2")
 
-        :param spec: spec of the package
-        :raises RuntimeError: in case of inconsistencies
-        """
-        error = 'you cannot ask for \'+{variant}\' when \'+mpi\' is not active'
+    # Cannot ask for +scalapack if it does not depend on MPI
+    conflicts('+scalapack', when='~mpi')
 
-        if '+scalapack' in spec and '~mpi' in spec:
-            raise RuntimeError(error.format(variant='scalapack'))
+    # Elpa is a substitute for scalapack and needs mpi
+    # conflicts('+elpa', when='~mpi')
```

Comment:
i know that `elpa` is not enabled yet, but maybe one can still keep those conflicts on so that we don't forget to add them later?